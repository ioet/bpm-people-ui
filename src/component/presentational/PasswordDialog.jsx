import React from 'react';
import Dialog from '@material-ui/core/Dialog/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import Button from '@material-ui/core/Button/Button';
import TextField from '@material-ui/core/TextField';
import Slide from '@material-ui/core/Slide/Slide';
import PropTypes from 'prop-types';
import { PasswordDialogConst, Variable } from '../../constants';

function Transition(props) {
  return <Slide direction="up" {...props} />;
}

const PasswordDialog = (props) => {
  const {
    open, inputError, password, passwordConfirm, handleClose, onChange,
  } = props;

  return (
    <Dialog
      open={open}
      TransitionComponent={Transition}
      keepMounted
      onClose={(e) => {
        e.preventDefault();
        handleClose(false);
      }}
    >
      <DialogTitle>
        {PasswordDialogConst.TITLE}
      </DialogTitle>
      <DialogContent>
        <DialogContentText>
          {PasswordDialogConst.EXPLANATION}
        </DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          name={Variable.PASSWORD}
          label={PasswordDialogConst.PASSWORD}
          value={password}
          type="password"
          onChange={
            (e) => {
              e.preventDefault();
              onChange(e);
            }
          }
          fullWidth
        />
        <form onSubmit={(e) => {
          e.preventDefault();
          handleClose(true);
        }}
        >
          <TextField
            error={inputError}
            margin="dense"
            name={Variable.PASSWORD_CONFIRM}
            label={PasswordDialogConst.CONFIRM_PASSWORD}
            value={passwordConfirm}
            type="password"
            onChange={
              (e) => {
                e.preventDefault();
                onChange(e);
              }
            }
            fullWidth
          />
        </form>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={(e) => {
            e.preventDefault();
            handleClose(false);
          }}
          color="primary"
        >
          {PasswordDialogConst.CANCEL}
        </Button>
        <Button
          onClick={(e) => {
            e.preventDefault();
            handleClose(true);
          }}
          color="primary"
        >
          {PasswordDialogConst.CREATE}
        </Button>
      </DialogActions>
    </Dialog>
  );
};

PasswordDialog.defaultProps = {
  inputError: false,
  password: '',
  passwordConfirm: '',
};

PasswordDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  inputError: PropTypes.bool,
  password: PropTypes.string,
  passwordConfirm: PropTypes.string,
  handleClose: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default PasswordDialog;
