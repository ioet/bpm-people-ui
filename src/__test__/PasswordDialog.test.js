import React from 'react';
import PasswordDialog from '../component/presentational/PasswordDialog';

describe('<PasswordDialog />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<PasswordDialog />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
