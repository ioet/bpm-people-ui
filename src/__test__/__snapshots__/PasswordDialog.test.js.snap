// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordDialog /> Component Renders correctly 1`] = `
<WithStyles(Dialog)
  TransitionComponent={[Function]}
  aria-describedby="alert-dialog-slide-description"
  aria-labelledby="alert-dialog-slide-title"
  keepMounted={true}
  onClose={[Function]}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-slide-title"
  >
    Create a new password for your account
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      Please choose a strong password with some special characters!
    </WithStyles(DialogContentText)>
    <TextField
      autoFocus={true}
      fullWidth={true}
      label="Enter a new password"
      margin="dense"
      name="password"
      onChange={[Function]}
      required={false}
      select={false}
      type="password"
      value=""
      variant="standard"
    />
    <TextField
      error={false}
      fullWidth={true}
      label="Please confirm your password"
      margin="dense"
      name="password_confirm"
      onChange={[Function]}
      required={false}
      select={false}
      type="password"
      value=""
      variant="standard"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Create
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
